extension/js/common/platform/store/contact-store.ts

```diff
@@ -105,24 +106,23 @@ export class ContactStore extends AbstractStore {
       }
       // X.509 certificate
       if (PgpKey.getKeyType(pubkey) === 'x509') {
-        // FIXME: For now we return random data.
-        // Later we'll return serial ID from the certificate.
-        const longid = Math.random() + '';
+        const fingerprint = smime.getSerialNumber(pubkey);
+        const longid = fingerprint.substring(fingerprint.length / 2);
```

Comment:
Is this following any known S/MIME spec or just trying to shoehorn it into our OpenPGP code?